pkg/api/api.go

```diff
@@ -54,6 +72,11 @@ func NewFromConfig(c KubernetesConfig) (Kubernetes, error) {
 		}
 	}
 
+	if discoveryClient != nil {
+		mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(discoveryClient))
```

Comment:
I'm not sure if we should use the memory cache here. I'm concerned about the impact in the memory footprint of tests.

Reply:
@pablochacin [DeferredDiscoveryRESTMapper](https://pkg.go.dev/k8s.io/client-go/restmapper#DeferredDiscoveryRESTMapper) is expecting [CachedDiscoveryInterface](https://pkg.go.dev/k8s.io/client-go@v0.26.0/discovery#CachedDiscoveryInterface). should i switch it to [disk.CachedDiscoveryClient](https://pkg.go.dev/k8s.io/client-go/discovery/cached/disk#CachedDiscoveryClient)? that seems to be the only other option i can find at the moment

Reply:
Let's keep it as it is now and then we can make some profiling.  Changing that seems an simple change